feature/security/gnsi/certz/client_certificates/README.md

```diff
@@ -0,0 +1,109 @@
+# X.X: gNSI Client Certificate Tests
+
+## Summary
+
+Clients must be able to validate a remote server's TLS certificate
+and present a valid client certificate to that server in order to provide
+identification information. The client certificate should have a
+SPIFFE Idenitifier embedded in it to be used as the identifier of
+the client to the server.
+
+## Baseline Setup
+
+### Input Args
+
+* the set of certificate testdata generated with the mk_cas.sh
+   script in featureprofiles/feature/security/certificate-authorties/
+
+### DUT service setup
+
+Configure the DUT to enable the following services (that are using gRPC) are
+up and user mTLS for authentication:
+
+* gNMI
+* gNOI
+* gNSI
+* gRIBI
+
+Be prepared to load the relevant trust_bundle.pem file for each test
+Certificate Authority(CA) under test on the DUT. Each CA has a RSA and ECDSA
+form, both must be tested.
+
+## Tests
+
+### Certz-1.0
+
+Perform these positive tests:
+
+Test that client certificates from a set of one CA are able to be validated and
+used for authentication to a device when used by a client connecting to each
+gRPC service.
+
+Perform this for both RSA and ECDSA signed CA bundles and
+certificates.
+Perform this for the permutations of 1, 2, 10, 1000 CA
+trust_bundle configurations: (## indicates the 1, 2, 10, 1000 CA testdata)
+
+   1) Load the correct key-type trust bundle onto the device and client system:
+        ca-##/trust_bundle_##_rsa.pem
+        ca-##/trust_bundle_##_ecdsa.pem
+
+   2) Load the correct key-type client certificate into the test gRPC client:
+        ca-##/client-rsa-key.pem
+        ca-##/client-rsa-cert.pem
+        ca-##/client-ecdsa-key.pem
+        ca-##/client-ecdsa-cert.pem
+
+   3) Load the correct key-type server certificate into the services on the DUT:
+        ca-##/server-rsa-key.pem
+        ca-##/server-rsa-cert.pem
+        ca-##/server-ecdsa-key.pem
+        ca-##/server-ecdsa-cert.pem
+
+   4) Validate that the certificate is loaded and useful for outbound
+      client connections.
+      
+   5) Connect to the service on the DUT.
+
+   6) Validate that the connection is established and that the client's
+      provided certificate is validated by the service on the DUT.
+
+   7) Validate that the connection's provided certificate is used for
```

Comment:
Any suggestions on how to validate it? Should we do something like: with the client cert_1 that has spiffe_id_wrong, expect the connection fails to establish; with client cert_2 that has spiffe_id_right, expect the connection can establish successfully?

Reply:
I don' t think the spiffe-id (itself) matters for the connection here.
The certificate/bundle matter. The device MAY require 'idX and not idY', but that's not this test (we'd have to specify that the service(s) on the device expect only 'idX', for instance - setup the authz and pathz parts).

This really is just basic 'tls-101', can you load certs and make connections that do not fail SSL/TLS primitive checks for the client->server direction:
  1) I know the signer (CA is in my bundle)
  2) the cert has a SAN/CN/SN for the thing I connected to.
  3) the certificate of the server is not expired.

The SSL/TLS client certificate sent to the device/server:
  1) known signer/CA at the server's view of the client cert
  2) client cert is not expired
  3) client-cert has a SPIFFE-ID in the SAN (though I think we don't even want to say this, because SPIFEE is mostly a google requirement right now, right?)

Reply:
and, for this we REALLY just care about the 2nd set of 3 things: "client certs presented to the server"